internal/updater/updater.go

```diff
@@ -23,54 +24,96 @@ import (
 	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/osutils"
 	"github.com/ActiveState/cli/internal/rtutils/ptr"
-	"github.com/gofrs/flock"
 )
 
-type ErrorInProgress struct{ *locale.LocalizedError }
+const (
+	CfgKeyInstallVersion = "state_tool_installer_version"
+	InstallerName        = "state-installer" + osutils.ExeExt
+)
 
-const CfgKeyInstallVersion = "state_tool_installer_version"
+type ErrorInProgress struct{ *locale.LocalizedError }
 
 var errPrivilegeMistmatch = errs.New("Privilege mismatch")
 
+type Origin struct {
+	Channel string
+	Version string
+}
+
+func NewOriginDefault() *Origin {
+	return &Origin{
+		Channel: constants.BranchName,
+		Version: constants.Version,
+	}
+}
```

Comment:
What purpose does this have? Far as I can tell Origin will ALWAYS be what we have in constants. State-svc isn't meant to address requests from any state tools other than the one it shipped with.

Reply:
So far, this is only for testing. Along with renaming the update construction functions, I added a comment to the one that ends up be used by the package level test.